devs/run-tests/25map.ts

```diff
@@ -0,0 +1,102 @@
+import * as ds from "@devicescript/core"
+import { assert } from "@devicescript/core"
+
+class Map<K, V> {
+    private map: { key: K; value: V }[] = []
+
+    size(): number {
+        return this.map.length
+    }
+
+    set(key: K, value: V): void {
+        const entry = this.getEntryByKey(key)
+        if (entry) {
+            entry.value = value
+        } else {
+            this.map.push({ key, value })
+        }
+    }
+
+    get(key: K): V | undefined {
+        const entry = this.getEntryByKey(key)
+        return entry ? entry.value : undefined
+    }
+
+    has(key: K): boolean {
+        return this.getEntryByKey(key) !== undefined
+    }
+
+    delete(key: K): boolean {
+        const entryIndex = this.findIndexByKey(key)
+        if (entryIndex !== -1) {
+            const newArray: { key: K; value: V }[] = []
+            for (let i = 0; i < this.map.length; i++) {
+                if (i !== entryIndex) {
+                    newArray.push(this.map[i])
+                }
+            }
+            this.map = newArray
+            return true
+        }
+        return false
+    }
+    
+    clear(): void {
+        this.map = []
+    }
```

Comment:
I was going to use Array splice, but it looks like i would need to open another PR to get that added to Array methods.